doc/introduction/pulse.rst

```diff
@@ -0,0 +1,289 @@
+leading to a time-dependent Hamiltonian interaction :math:`H(t)`. We call driving the system with such an
+electromagnetic field for a fixed time window a *pulse sequence*. These pulse sequences can then be tuned to
+implement the higher level gates used for quantum computation.
+
+The :mod:`~.pulse` module provides functions and classes used to simulate pulse-level control of quantum
+systems. It contains a :class:`~.ParametrizedHamiltonian` class, which can be used to define the time-dependent
+Hamiltonian describing the interaction between the applied pulses and the system. The
+:class:`~.ParametrizedHamiltonian` can be used to create a :class:`~.ParametrizedEvolution`
+(:class:`~.Operation`) that provides the time evolution of the Hamiltonian. The :mod:`~.pulse` module also
+includes several convenience functions for defining pulse envelopes.
+
+The :mod:`~.pulse` module relies on the external package `JAX <https://jax.readthedocs.io/en/latest/>`_, which
+requires separate installation. The module is written for ``jax`` and will not work with the other machine learning
+frameworks typically encountered in PennyLane.
+
+
+
+Creating a parametrized Hamiltonian
+-----------------------------------
+
+The :mod:`~.pulse` module provides a framework to create a time-dependent Hamiltonian of the form
+
+.. math:: H(p, t) = H_\text{drift} + \sum_j f_j(p, t) H_j
+
+with constant operators :math:`H_j` and scalar functions :math:`f_j(p, t)` that may depend on
+parameters :math:`p` and time :math:`t`.
+
+.. note::
+    The :class:`~.ParametrizedHamiltonian` is not a PennyLane :class:`~.Operator`. If an :class:`~.Operator`
+    representing the :class:`~.ParametrizedHamiltonian` is needed, the initialized :class:`~.ParametrizedHamiltonian`
+    must be called with fixed parameters and time.
+
+Defining a :class:`~.ParametrizedHamiltonian` requires coefficients and operators. In the example below, we define a
+Hamiltonian with a single drift term, and two parametrized terms.
+
+.. code-block:: python
+
+    import pennylane as qml
+    from jax import numpy as jnp
+
+    # defining the coefficients fj(v, t) for the two parametrized terms
+    f1 = lambda p, t: p * jnp.sin(t) * (t - 1)
+    f2 = lambda p, t: p[0] * jnp.cos(p[1]* t ** 2)
+
+    # defining the operations for the three terms in the Hamiltonian
+    XX = qml.PauliX(0) @ qml.PauliX(1)
+    YY = qml.PauliY(0) @ qml.PauliY(1)
+    ZZ = qml.PauliZ(0) @ qml.PauliZ(1)
+
+The functions defining the parameterized coefficients must have the call signature ``(p, t)``, where ``p`` can be a float,
+list or jnp.array. These functions should be defined using ``jax.numpy`` rather than ``numpy`` where relevant.
+
+There are two way to construct a :class:`~.ParametrizedHamiltonian` from the coefficients
+and operators:
+
+.. code-block:: python
+
+    # Option 1
+    H1 =  2 * XX + f1 * YY + f2 * ZZ
+
+    # Option 2
+    coeffs = [2, f1, f2]
+    ops = [XX, YY, ZZ]
+    H2 =  qml.ops.dot(coeffs, ops)
```

Comment:
We will use `qml.dot` when deprecating `qml.collections.dot`. Either we change it now, or we set a reminder to change it before the release

```suggestion
    H2 =  qml.dot(coeffs, ops)
```